modules/videoeditor-ffmpeg-impl/src/main/java/org/opencastproject/videoeditor/impl/VideoEditorServiceImpl.java

```diff
@@ -279,19 +278,88 @@ protected Track processSmil(Job job, Smil smil, String trackParamGroupId) throws
           }
         }
       }
-      List<VideoClip> cleanclips = sortSegments(videoclips);    // remove very short cuts that will look bad
-      String error = null;
-      String outputResolution = "";    //TODO: fetch the largest output resolution from SMIL.head.layout.root-layout
-      // When outputResolution is set to WxH, all clips are scaled to that size in the output video.
-      // TODO: Each clips could have a region id, relative to the root-layout
-      // Then each clip is zoomed/panned/padded to WxH befor concatenation
-      FFmpegEdit ffmpeg = new FFmpegEdit(properties);
-      error = ffmpeg.processEdits(inputfile, outputPath.getAbsolutePath(), outputResolution, cleanclips,
-              sourceTrack.hasAudio(), sourceTrack.hasVideo());
-
-      if (error != null) {
-        FileUtils.deleteQuietly(tempDirectory);
-        throw new ProcessFailedException("Editing pipeline exited abnormaly! Error: " + error);
+
+      // Don't mix video/audio and subtitles
+      if (videoclips.size() > 0 && refElements.size() > 0) {
+        throw new ProcessFailedException("Can not process media elements together with ref elements. "
+                + "There likely is an error in the SMIL file");
+      }
```

Comment:
Been a bit, but if I remember correctly: According to official SMIL definition no, `refs` are not subtitles, but just one of the synonyms for media elements (https://www.w3.org/TR/REC-smil/smil-extended-media-object.html#edef-ref). BUT in Opencast, refs are basically anything that are not video or audio, and the videoeditor fails if it finds one. That's also why we later check if the ref element links to a `.vtt` file, since that is the only kind of ref element we are equippped to process.